pkg/controller/keycloak/keycloak_migrations.go

```diff
@@ -37,10 +38,45 @@ func GetMigrator(cr *v1alpha1.Keycloak) (Migrator, error) {
 }
 
 func (i *RecreateMigrator) Migrate(cr *v1alpha1.Keycloak, currentState *common.ClusterState, desiredState common.DesiredClusterState) (common.DesiredClusterState, error) {
-	if needsMigration(cr, currentState) {
+	deployment, deploymentIndex := findDeployment(&desiredState)
+
+	// We can't modify existing selector on StatefulSet.
+	// The selector might be wrongly set by e.g. RH-SSO 7.5.2.
+	// In such case, we need to recreate the StatefulSet
```

Comment:
This looks to be a super dangerous operation, especially combined with OLM automatic updates: https://github.com/kubernetes/kubernetes/issues/50808#issuecomment-324746374

Would it be possible only to notify the user (e.g. using the status field) and leave the deletion to be a manual action performed by the user, so that it can be better scheduled/executed in production scenarios?